ansible_collections/arista/cvp/molecule/cv_device_v3/test_move_and_deploy_device.yml

```diff
@@ -0,0 +1,160 @@
+      arista.cvp.cv_container_v3:
+        topology: '{{CONTAINER}}'
+
+    - name: Collect devices facts from {{inventory_hostname}}
+      arista.cvp.cv_facts_v3:
+        facts:
+          - devices
+        regexp_filter: "{{DEVICE_NAME}}"
+      register: CV_FACTS_V3_RESULT
+
+    - name: Run CV_DEVICE_V3 To Move Device
+      arista.cvp.cv_device_v3:
+        devices: '{{CVP_DEVICES_MOVE_DEVICE}}'
+        state: present
+      register: MOVE_DEVICE_RUN
+
+    - name: Pause for 10 seconds to Execute Task
+      ansible.builtin.pause:
+        seconds: 10
+
+    - name: Execute Task for deploy_Device
+      arista.cvp.cv_task_v3:
+        tasks: "{{MOVE_DEVICE_RUN.taskIds}}"
+
+    - name: Check Move Device
+      assert:
+        that:
+          - "MOVE_DEVICE_RUN.devices_moved.changed == true"
+          - "MOVE_DEVICE_RUN.devices_moved.devices_moved_count == 1"
+          - "MOVE_DEVICE_RUN.devices_moved.devices_moved_list == ['leaf1_to_Test_Leaf']"
+          - "MOVE_DEVICE_RUN.devices_moved.success == true"
+          - "MOVE_DEVICE_RUN.failed == false"
+
+    - name: Run CV_DEVICE_V3 To Move Device With Parent Container None
+      arista.cvp.cv_device_v3:
+        devices: '{{CVP_DEVICES_PARENT_CONTAINER_NONE}}'
+        state: present
+      ignore_errors: yes
+      register: MOVE_DEVICE_NONE
+
+    - name: Negative Test Check Move Device With Parent Container None
+      assert:
+        that:
+          - MOVE_DEVICE_NONE.msg == "The target container 'None' for the device 'leaf1' does not exist on CVP."
+          - MOVE_DEVICE_NONE.changed == false
+          - MOVE_DEVICE_NONE.failed == true
+
+    - name: Run CV_DEVICE_V3 To Move Device With Parent Container Same
+      arista.cvp.cv_device_v3:
+        devices: '{{CVP_DEVICES_MOVE_DEVICE}}'
+        state: present
+      ignore_errors: yes
+      register: MOVE_DEVICE
+
+    - name: Negative Test Check Move Device With Same Parent Container
+      assert:
+        that:
+          - MOVE_DEVICE.devices_moved.changed == false
+          - MOVE_DEVICE.devices_moved.devices_moved_count == 0
+          - MOVE_DEVICE.devices_moved.devices_moved_list == []
+          - MOVE_DEVICE.devices_moved.success == false
+          - MOVE_DEVICE.failed == false
+          - MOVE_DEVICE.taskIds: []
+
+    ################################
+    #        DEPLOY DEVICE        ##
+    ################################
+
+     # moving device to undefined container
+    - name: Run CV_DEVICE_V3 With State Provisioning Reset
+      arista.cvp.cv_device_v3:
+        devices: '{{CVP_DEVICES_PROVISIONING_RESET}}'
+        state: provisioning_reset
+      register: PROVISIONING_RESET_RESULT
+
+    - name: Pause for 10 seconds to Execute Task
+      ansible.builtin.pause:
+        seconds: 10
+
+    - name: Validate configurations
+      arista.cvp.cv_validate_v3:
+        devices: "{{CVP_DEVICES_VALIDATE_CONFIG}}"
+        validate_mode: stop_on_error
+      register: CVP_DEVICES_RESULTS
```

Comment:
Why are we validating the configs here? We have molecule tests for cv_validate_v3 separately.

Reply:
Tamas suggested to verify the configlets before applying it in deploy_device.